__tests__/run.test.ts

```diff
@@ -1,7 +1,104 @@
+        jest.spyOn(fs, 'readFileSync').mockReturnValue('v1.20.4');
+        
+        expect(await run.getStableKubectlVersion()).toBe('v1.20.4');
+        expect(toolCache.downloadTool).toBeCalled();
+        expect(fs.readFileSync).toBeCalledWith('pathToTool', 'utf8');
+    });
+
+    test('getStableKubectlVersion() - return default v1.15.0 if version read is empty', async () => {
+        jest.spyOn(toolCache, 'downloadTool').mockReturnValue(Promise.resolve('pathToTool'));
+        jest.spyOn(fs, 'readFileSync').mockReturnValue('');
+        
+        expect(await run.getStableKubectlVersion()).toBe('v1.15.0');
+        expect(toolCache.downloadTool).toBeCalled();
+        expect(fs.readFileSync).toBeCalledWith('pathToTool', 'utf8');
+    });
+
+    test('getStableKubectlVersion() - return default v1.15.0 if unable to download file', async () => {
+        jest.spyOn(toolCache, 'downloadTool').mockRejectedValue('Unable to download.');
+        
+        expect(await run.getStableKubectlVersion()).toBe('v1.15.0');
+        expect(toolCache.downloadTool).toBeCalled();
+    });
+
+    test('downloadKubectl() - download kubectl, add it to toolCache and return path to it', async () => {
+        jest.spyOn(toolCache, 'find').mockReturnValue('');
+        jest.spyOn(toolCache, 'downloadTool').mockReturnValue(Promise.resolve('pathToTool'));
+        jest.spyOn(toolCache, 'cacheFile').mockReturnValue(Promise.resolve('pathToCachedTool'));
+        jest.spyOn(os, 'type').mockReturnValue('Windows_NT');
+        jest.spyOn(fs, 'chmodSync').mockImplementation(() => {});
+        
+        expect(await run.downloadKubectl('v1.15.0')).toBe('pathToCachedTool/kubectl.exe');
+        expect(toolCache.find).toBeCalledWith('kubectl', 'v1.15.0');
+        expect(toolCache.downloadTool).toBeCalled();
+        expect(toolCache.cacheFile).toBeCalled();
+        expect(os.type).toBeCalled();
+        expect(fs.chmodSync).toBeCalledWith('pathToCachedTool/kubectl.exe', '777');
+    });
+
+    test('downloadKubectl() - throw DownloadKubectlFailed error when unable to download kubectl', async () => {
+        jest.spyOn(toolCache, 'find').mockReturnValue('');
+        jest.spyOn(toolCache, 'downloadTool').mockRejectedValue('Unable to download kubectl.');
+
+        await expect(run.downloadKubectl('v1.15.0')).rejects.toThrow('DownloadKubectlFailed');
+        expect(toolCache.find).toBeCalledWith('kubectl', 'v1.15.0');
+        expect(toolCache.downloadTool).toBeCalled();
+    });
+
+    test('downloadKubectl() - return path to existing cache of kubectl', async () => {
+        jest.spyOn(toolCache, 'find').mockReturnValue('pathToCachedTool');
+        jest.spyOn(os, 'type').mockReturnValue('Windows_NT');
+        jest.spyOn(fs, 'chmodSync').mockImplementation(() => {});
+        
+        expect(await run.downloadKubectl('v1.15.0')).toBe('pathToCachedTool/kubectl.exe');
+        expect(toolCache.find).toBeCalledWith('kubectl', 'v1.15.0');
```

Comment:
You can add an `expect` here to check that toolCache.downloadTool doesn't get called using `.not.toBeCalled`.